.github/workflows/release.yml

```
@@ -344,12 +335,137 @@ jobs:
+          echo "PREVIOUS_FULL_RELEASE_TAG=$latest_release_tag" >> $GITHUB_ENV
+      - name: Fetch Reference Binary
+        uses: robinraju/release-downloader@768b85c8d69164800db5fc00337ab917daf3ce68
+        with:
+          tag: ${{env.PREVIOUS_FULL_RELEASE_TAG}}
+          fileName: ${{env.BIN_FILENAME}}
+          out-file-path: ${{env.REF_DOWNLOAD_DIR}}
+      - name: Download Test Binary
+        uses: actions/download-artifact@v3
+        with:
+          name: artifacts-${{github.run_id}}
+          path: ${{env.TEST_DOWNLOAD_DIR}}
+      - name: Rename and Move Binaries
+        run: |
+          mkdir -p ${{env.BIN_DIR}}
+          mv ${{env.REF_DOWNLOAD_DIR}}/${{env.BIN_FILENAME}} ${{env.BIN_DIR}}/${{env.REF_BIN_FILENAME}}
+          mv ${{env.TEST_DOWNLOAD_DIR}}/${{env.BIN_FILENAME}} ${{env.BIN_DIR}}/${{env.TEST_BIN_FILENAME}}
+      - name: Set Binary Permissions
+        working-directory: ${{env.BIN_DIR}}
+        run: |
+          set -x
+          chmod 755 $TEST_BIN_FILENAME
+          chmod 755 $REF_BIN_FILENAME
+      - name: Start Test Node
+        working-directory: ${{env.BIN_DIR}}
+        run: |
+          ./${{env.TEST_BIN_FILENAME}} \
+            --rpc-external \
+            --rpc-cors=all \
+            --ws-external \
+            --rpc-methods=Unsafe \
+            --no-telemetry \
+            --no-prometheus \
+            --port $((30333)) \
+            --rpc-port $((9933)) \
+            --ws-port $((9944)) \
+            --reserved-only \
+            --tmp \
+            -- \
+            --reserved-only \
+            &
+      - name: Start Reference Node
+        working-directory: ${{env.BIN_DIR}}
+        run: |
+          ./${{env.REF_BIN_FILENAME}} \
+            --rpc-external \
+            --rpc-cors=all \
+            --ws-external \
+            --rpc-methods=Unsafe \
+            --no-telemetry \
+            --no-prometheus \
+            --port $((30334)) \
+            --rpc-port $((9934)) \
+            --ws-port $((9945)) \
+            --reserved-only \
+            --tmp \
+            --  \
+            --reserved-only \
+            &
+      - name: Prepare Output
+        working-directory: ${{env.OUTPUT_DIR}}
+        run: |
+          REF_VERSION=$(./${{env.REF_BIN_FILENAME}} --version)
+          BIN_VERSION=$(./${{env.TEST_BIN_FILENAME}} --version)
+          echo "Metadata comparison against previous full release:" >> ${{env.OUTPUT_FILENAME}}
+          echo "Date: $(date)" >> ${{env.OUTPUT_FILENAME}}
+          echo "Ref. version: $REF_VERSION" >> ${{env.OUTPUT_FILENAME}}
+          echo "Test version: $BIN_VERSION" >> ${{env.OUTPUT_FILENAME}}
+          echo "----------------------------------------------------------------------" >> ${{env.OUTPUT_FILENAME}}
+      - name: Compare Metadata
+        run: |
+          CMD="docker run --pull always --network host jacogr/polkadot-js-tools metadata ws://localhost:9945 ws://localhost:9944"
```

Here's the other side of that port problem.
```suggestion
          CMD="docker run --pull always --network host jacogr/polkadot-js-tools metadata ws://localhost:9946 ws://localhost:9944"
```